operator/internal/handlers/internal/gateway/tenant_secrets.go

```diff
@@ -71,10 +107,27 @@ func extractSecret(s *corev1.Secret, tenantName string) (*manifests.TenantSecret
 	clientSecret := s.Data["clientSecret"]
 	issuerCAPath := s.Data["issuerCAPath"]
 
-	return &manifests.TenantSecrets{
-		TenantName:   tenantName,
+	return &manifests.OIDCSecret{
 		ClientID:     string(clientID),
 		ClientSecret: string(clientSecret),
 		IssuerCAPath: string(issuerCAPath),
 	}, nil
 }
+
+// extractTLSSecret reads a k8s secret into a manifest tenant secret struct if valid.
+func extractTLSSecret(s *corev1.Secret, tenantName string) (*manifests.TLSSecret, error) {
+	// Extract and validate mandatory fields
+	cert := s.Data["cert"]
+	if len(cert) == 0 {
+		return nil, kverrors.New("missing cert field", "field", "cert")
+	}
+	ca := s.Data["ca"]
+	if len(ca) == 0 {
+		return nil, kverrors.New("missing ca field", "field", "ca")
+	}
```

Comment:
I would consider to require that the Secret is of the following type `kubernetes.io/tls` that make handling the keys more uniform similar to other secrets containing TLS certs/keys (i.e. pkg `certrotation`). 

Excerpt from `"k8s.io/api/core/v1"`:
```go
	// SecretTypeTLS contains information about a TLS client or server secret. It
	// is primarily used with TLS termination of the Ingress resource, but may be
	// used in other types.
	//
	// Required fields:
	// - Secret.Data["tls.key"] - TLS private key.
	//   Secret.Data["tls.crt"] - TLS certificate.
	// TODO: Consider supporting different formats, specifying CA/destinationCA.
	SecretTypeTLS SecretType = "kubernetes.io/tls"

	// TLSCertKey is the key for tls certificates in a TLS secret.
	TLSCertKey = "tls.crt"
	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
	TLSPrivateKeyKey = "tls.key"
```

In a further note I would put the CA in a separate configmap to handle it like a CA bundle. In details CAs are exported or generate in a list (including the intermediate CAs) and have a long-lived rotation than server certificates issued by those. We can but might build a similar struct like:

```go
// ObjectStorageTLSSpec is the TLS configuration for reaching the object storage endpoint.
type ObjectStorageTLSSpec struct {
	// Key is the data key of a ConfigMap containing a CA certificate.
	// It needs to be in the same namespace as the LokiStack custom resource.
	// If empty, it defaults to "service-ca.crt".
	//
	// +optional
	// +kubebuilder:validation:optional
	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="CA ConfigMap Key"
	CAKey string `json:"caKey,omitempty"`
	// CA is the name of a ConfigMap containing a CA certificate.
	// It needs to be in the same namespace as the LokiStack custom resource.
	//
	// +required
	// +kubebuilder:validation:required
	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:ConfigMap",displayName="CA ConfigMap Name"
	CA string `json:"caName"`
}
```